teachertool/src/App.tsx

```diff
@@ -30,13 +32,40 @@ function App() {
         }
     }, [ready]);
 
+    const onIframeLoaded = () => {
+        console.log("iframe loaded");
+    }
+
+    const onIframeClosed = () => {
+        console.log("iframe closed");
+    }
+
+    const createIFrameUrl = (): string => {
```

Comment:
No need to block on this, but as a nit-pick, I think app.tsx may not be the right place for this function. I'd recommend maybe putting it inside a util function or perhaps even inside the MakeCodeFrame file...but since you're trying to make the MakeCodeFrame more generic, I'd lean towards util (or even creating an additional component that wraps the MakeCodeFrame...)

Reply:
Yeah, that's a good point. I'll put the frame in a separate component. I think this will be better for future use, anyway, since I would like to lift the generic frame to react-common in the near future.